doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -471,6 +471,37 @@ Field              | Type       | Meaning
 `cpu_nano_cores`   | [`bigint`] | Approximate CPU usage, in billionths of a vCPU core.
 `memory_bytes`     | [`bigint`] | Approximate RAM usage, in bytes.
 
+### `mz_source_statistics`
+
+The `mz_storage_source_statistics` table contains statistics for each process of each source in the system.
+
+Materialize does not make any guarantees about the exactness or freshness of these numbers. As internal
+components of the system are restarted, some of these metrics will be reset to 0.
+
+Field                 | Type         | Meaning
+----------------------|--------------|--------
+`id`                  | [`text`]     | The ID of the source.
+`worker_id`           | [`bigint`]   | An identifier of a worker within the source.
+`snapshot_committed`  | [`boolean`]  | Whether or not the worker has committed the initial snapshot for a source.
+`messages_received`   | [`bigint`]   | How many raw messages the worker has received from upstream.
+`updates`             | [`bigint`]   | How many updates (inserts + deletes) the worker has written but not yet committed to the storage layer.
+`updates_committed`   | [`bigint`]   | How many updates (inserts + deletes) the worker has committed into our storage layer.
+`bytes_received`      | [`bigint`]   | How many bytes worth of messages the worker has received from upstream. The way the bytes are counted is source-specific.
+
+These values all require some amount of interpretation:
+
+- `messages_received` does not distinguish between inserts and deletes. Some upstream sources, like postgres,
+  produce deletion messages that are counted the same as insertions.
+- The `updates*` metrics can be permanently lower or higher than `messages_received`, because:
+  - The storage layer can choose to consolidate messages within a certain time interval.
+  - Some envelopes (like `UPSERT`) can produce 2 messages (an insert and a delete) from a single upstream message.
+- The `messages_*` and `bytes_received` metrics can optionally be interpreted as rates that can loosely show
+  the changing speed of a source over time.
+- In general, the values should be summed across `worker_id`'s, and some metrics may only exist for some workers.
+  - Additionally, sources with multiple outputs (like postgres shards) will need `messages_committed`
+  and `snapshot_committed` summed/AND-ed across the top-level source and the connected tables.
```

Comment:
```suggestion
- `messages_received` does not distinguish between inserts and deletes. Some upstream sources, like PostgreSQL,
  produce deletion messages that are counted the same as insertions.
- The `updates*` metrics can be permanently lower or higher than `messages_received`, because:
  - The storage layer can choose to consolidate messages within a certain time interval.
  - Some envelopes (like `UPSERT`) can produce 2 messages (an insert and a delete) from a single upstream message.
- The `messages_*` and `bytes_received` metrics can optionally be interpreted as rates that can loosely show
  the changing speed of a source over time.
- In general, the values should be summed across `worker_id`s, and some metrics may only exist for some workers.
  - Additionally, sources with multiple outputs (like PostgreSQL) will need `messages_committed`
  and `snapshot_committed` summed/AND-ed across the top-level source and its subsources.
```

Reply:
It's reasonable to keep these details here, for lack of a better place (at least for the time being).